ssh/tailssh/tailssh.go

```diff
@@ -160,62 +160,105 @@ func (srv *server) handleSSH(s ssh.Session) {
 	if !action.Accept || action.HoldAndDelegate != "" {
 		fmt.Fprintf(s, "TODO: other SSHAction outcomes")
 		s.Exit(1)
-
 	}
-	if !isPty {
-		fmt.Fprintf(s, "TODO scp etc\n")
+	lu, err := user.Lookup(localUser)
+	if err != nil {
+		logf("ssh: user Lookup %q: %v", localUser, err)
 		s.Exit(1)
-		return
 	}
+
+	logf("ssh: connection from %v %v to %v@ => %q. command = %q, env = %q", srcIP, uprof.LoginName, sshUser, localUser, s.Command(), s.Environ())
 	var cmd *exec.Cmd
-	if os.Getuid() != 0 {
-		u, err := user.Current()
-		if err != nil {
-			logf("failed to get current user: %v", err)
+	if euid := os.Geteuid(); euid != 0 {
+		if lu.Uid != fmt.Sprint(euid) {
+			logf("ssh: can't switch to user %q from process euid %v", localUser, euid)
+			fmt.Fprintf(s, "can't switch user\n")
 			s.Exit(1)
 			return
 		}
-		if u.Username != localUser {
-			fmt.Fprintf(s, "can't switch user\n")
+		cmd = exec.Command(loginShell(lu.Uid))
+	} else {
+		if rawCmd := s.RawCommand(); rawCmd != "" {
+			cmd = exec.Command("/usr/bin/env", "su",
+				"-c", rawCmd,
+				localUser)
+			cmd.Dir = lu.HomeDir
+			cmd.Env = append(cmd.Env,
```

Comment:
pull this out into a func like you did for loginShell

Reply:
done